index.js

```diff
@@ -366,9 +368,23 @@ Archive.prototype.appendFile = function (filename, name, cb) {
       mode: st.mode,
       size: 0,
       link: null
-    }, {filename: filename}, cb)
+    }, {filename: filename}, function (err, entry) {
+      if (err) return cb(err)
+      if (!entry) return cb()
+      if (entry.type === 'file') self.readStats.files++
+      cb(err, entry)
+    })
 
-    if (ws) pump(fs.createReadStream(path.resolve(self.directory, filename)), ws)
+    var statsCounter = through(function (data, encoding, next) {
+      self.readStats.totalRead += data.length
+      this.push(data)
+      next()
+    })
+
+    if (ws) {
+      var readStream = fs.createReadStream(path.resolve(self.directory, filename))
+      pump(readStream, statsCounter, ws)
+    }
```

Comment:
@joehand progress api:

``` js
var stats = drive.appendFile(..., function () {
  clearInterval(interval)
  console.log('File added')
})

var interval = setInterval(function () {
  console.log(stats.bytesRead + ' / ' + stats.bytesTotal)
}, 1000)
```

@maxogden yes i agree that a stats api is a good idea in general but if it tracks how many files were added it should track the other types as well, https://github.com/mafintosh/hyperdrive/blob/master/index.js#L377-L387.